### DISCUSSION TITLE
Create Event to hook-into VAT validation.

### DISCUSSION THREAD

**Discussion by lenvanessen** [craftcms/commerce#2465] [Ideas]
**What are you trying to do?**
We want to be able to hook into the VAT validation to be able to add some additional business logic. In our case:
- Only validate the number if the country is not equal to the store country
- Check if the VAT-code matches with the given country of shipment, rather then allowing people to use VAT-codes beloning to other countries.
-
**What's your proposed solution?**
Drop-in a event somewhere in the VAT validation process.

**Comment by lenvanessen** [comment]
Or allow a custom VAT validator to be used, rather than the default one

**Comment by lenvanessen** [comment]
Hi Guys,

We urgently need this, currently there is no way to properly adjust the vat validation logic, and it's flawed. We have people placing orders with 0% vat, with a shipping country set to Germany and using a Belgium VAT number. While it is indeed a valid VAT number, it does not meet the requirements for EU vat exception.

It should be really easy to make this extendable, by event or a configuration for a custom vat validation class. Pretty please, can we have an event?

**Comment by lenvanessen** [comment]
@lukeholder @brandonkelly could anyone pick this up? Our client's left paying the taxes for people fraudulently using foreign tax numbers that don't match the country where the bill/goods are going to. It's been three years and no movement here, how can we get this moving along? Would a PR help? If yes, what approach? Adding an event when the vat validation is done to override the result?

**Comment by lukeholder** [comment]
@lenvanessen sorry for the delay on this. I am currently working on moving the tax adjuster to use the Vat service, and we will add an event there to add additional logic for validation of Vat IDs, including checking the address. I will update here when it is ready for testing. This will likely be in a 5.1 beta. Thanks

**Comment by WHITE-developer** [comment]
Hi @lukeholder, isn't it possible to add this in commerce 4 as well? We also have a client having problems with this.

**Comment by lukeholder** [comment]
@WHITE-developer I will look to add it to Commerce 4 also.
